Implement periodic timer for querying OTA providers, refactor OTARequestor and OTARequestorDriver (#15433)
project-chip/connectedhomeip: examples/ota-requestor-app/linux/README.md

| Directory                         | Description                                                                                                                                                                                                                                          |
| --------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | -p/--periodicQueryTimeout <Time in seconds> | Timeout for querying providers on the default OTA provider list. If none or zero is supplied the timeout is set to 24 hours  |

Reviewer: Small suggestion to make it clear this value is periodic

```suggestion
| -p/--periodicQueryTimeout <Time in seconds> | Periodic timeout for querying providers in the default OTA provider list. If none or zero is supplied the timeout is set to every 24 hours.  |
```

-   `${KVS_STORE_LOCATION}` is a location where the KVS items will be stored. If
    none is supplied, the default is /tmp/chip_kvs. This must be different from
    the value used by the OTA Provider application.

> -   `${TIME_IN_SECONDS}` is the imeout for querying providers on the default OTA
>     provider list. If none or zero is supplied the timeout is set to 24 hours

Reviewer: ```suggestion
-   `${TIME_IN_SECONDS}` is the periodic timeout for querying providers in the default OTA provider list. If none or zero is supplied the timeout is set to every 24 hours.
```
